progress/README.md

```diff
@@ -0,0 +1,305 @@
+|     [POST /progresses](#post-progresses)      |                                 Creates a new progress entry.                                 |
+|      [GET /progresses](#get-progresses)       |                 Retrieves progress entries based on the provided parameters.                  |
+| [GET /progresses/range](#get-progressesrange) | Retrieves the progress records for a particular user or task within the specified date range. |
+
+## POST /progresses
+
+    Creates a new progress entry.
+
+- **Params**  
+  None
+- **Query**
+  None
+- **Body**  
+  Attributes:
+  -type (required, string): Specifies the type of the progress entry (e.g., "task", "user").
+  -taskId (optional, string): Task ID associated with the progress entry (applicable for task progress only).
+  -completed (required, string): Completed portion of the task or progress.
+  -planned (required, string): Planned portion of the task or progress.
+  -blockers (required, string): Blockers for the progress entry.
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+  - **Code:** 201
+    - **Content:**
+      ```json
+      {
+        "message": "String",
+        "data": {
+          "type": "String",
+          "completed": "String",
+          "planned": "String",
+          "blockers": "String",
+          "userId": "String",
+          "taskId": "String (for task type)",
+          "createdAt": "Timestamp",
+          "date": "Timestamp",
+          "id": "String"
+        }
+      }
+      ```
+- **Error Response:**
+
+  - **Code:** 401
+    - **Content:**
+      ```json
+      {
+        "statusCode": 401,
+        "error": "Unauthorized",
+        "message": "Unauthenticated User."
+      }
+      ```
+  - **Code:** 400
+
+    - **Content:**
+      ```json
+      {
+        "message": "Bad Request"
+      }
+      ```
+
+  - **Code:** 500
+    - **Content:**
+      ```json
+      {
+        "message": "Internal Server Error"
+      }
+      ```
+
+- **Example for user progress creation request:**
+  POST /progresses
+  Content-Type: application/json
+  Request-Body:
+
+  ```json
+  {
+    "type": "user",
+    "completed": "Monitoring for issues and performance",
+    "planned": "Plan for future enhancements",
+    "blockers": "None"
+  }
+  ```
+
+  Response :
+  Status 201
+  Content-Type: application/json
+
+  ```json
+  {
+    "data": {
+      "type": "user",
+      "completed": "Monitoring for issues and performance",
+      "planned": "Plan for future enhancements",
+      "blockers": "None",
+      "userId": "SooJK37gzjIZfFNH0tlL",
+      "createdAt": 1683688367555,
+      "date": 1683676800000,
+      "id": "t5k77PHnuDSrgEzvMJAj"
+    },
+    "message": "User Progress document created successfully."
+  }
+  ```
+
+  for progress document already created
+  Status 400
```

Comment:
I think this request should be a put request and re-creation of data should return a 409 status code

Reply:
Given that the intention is to create a new document and there is no allowance for users to update the record after creation, it would be more appropriate to use the HTTP POST method.

Reply:
Thanks for pointing out I will add support for returning 409 Conflict when progress document already exists.

Reply:
As per mdn here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT
Put is used for the creation of a unique new resource
![image](https://github.com/Real-Dev-Squad/website-api-contracts/assets/57758447/276b932a-917c-4b93-b45b-e9c9ffa821be)

Reply:
`The difference between PUT and POST is that PUT is idempotent: calling it once or several times successively has the same effect (that is no side effect), where successive identical POST may have additional effects, like passing an order several times.` 
For us, calling the API multiple times will not have the same effect. Once the document for the day is created by calling the API for the first time, subsequent consecutive calls will throw an error. Hence POST for creating progress documents in our case would be appropriate because it allows for non-idempotent creation of a new resource.